freqtrade/commands/arguments.py

```diff
@@ -212,6 +212,16 @@
 ARGS_HYPEROPT_SHOW = [
     "hyperopt_list_best",
     "hyperopt_list_profitable",
+    "hyperopt_list_min_trades",
+    "hyperopt_list_max_trades",
+    "hyperopt_list_min_avg_time",
+    "hyperopt_list_max_avg_time",
+    "hyperopt_list_min_avg_profit",
+    "hyperopt_list_max_avg_profit",
+    "hyperopt_list_min_total_profit",
+    "hyperopt_list_max_total_profit",
+    "hyperopt_list_min_objective",
+    "hyperopt_list_max_objective",
```

Comment:
to understand this right - you're adding random parameters to a command which are then ignored?

from a usability perspective - i think that's pretty bad - as it makes it totally unclear which of the 20 parameters actually do something.

Reply:
as far as i could tell this 

the list/show command use the same filter mechanism 

https://github.com/freqtrade/freqtrade/blob/f3d232563dc9c3ee311001144ea81d50db1a29a8/freqtrade/commands/hyperopt_commands.py#L79
and here
https://github.com/freqtrade/freqtrade/blob/f3d232563dc9c3ee311001144ea81d50db1a29a8/freqtrade/commands/hyperopt_commands.py#L34

but the show command actually ditches the filter options since they are not passed/evaluated
if you just add them there already they will be used by the `HyperoptTools.load_filtered_results(results_file, config)`
since it passes always the complete config to that command 


in this case it could even maybe make sense to combine those functions 
and say if you pass an -n or -epoch to the list function you could get the actual result

Reply:
ah - so it only saves memory IF additional parameters are passed in - but if i run `hyperopt-show -n 1` it doesn't?

Well, we could have `load_filtered_results` evaluate the `-n` option, too, in that case - so the epochs would **actually** not be loaded (with `-n 1` - we only need to load one epoch) ?

Reply:
yeah I gues this would also be possible then from usubilty perspective we could have one command 

e.g. `hyperopt-list/show [args] -n <epoch>`
where essentially we could say in the docs that 

"-n: get show result from the single epoch of the filtered list"

I can have a look maybe on monday onto this how to evealute the -n from the `load_filtered_results`

Reply:
- i don't think allowing `-n` on list makes sense. The "list" command is to compare results - not to get an individual one.

Reply:
We could also add a specific query command uses most of the inputs from the list and also allows to query for the current_epoch field 
With this we would not really need to change any of the existing functions

Reply:
that was my idea, really 😆 

just move "hyperopt_show_index" into the function - with a default that you ignore if it's not set (which would be the case for a "hyperopt-list" call.
it'll then only return one value - if it's a low number - there's no need to fetch all subsequent epochs - which is what we're doing currently.

Reply:
reconsidering how hyperopt-show works - i don't think this approach will work (letalone save memory).

`-n` is the index of the hyperopt result - so `-2` will get you the result 2nd to last.
The only way to save memory is by short-circuiting the filters - which your current approach doesn't do - and it really can't do unless the index is positive.

Even with that - it's the index ... so for a combination as outlined in [the docs](https://www.freqtrade.io/en/latest/utils/#examples_3) (see below) - you still need to always load all epochs - as otherwise you'll never know if "another" "best" is comming along.

> Prints JSON data with details for the last best epoch (i.e., the best of all epochs):
> `freqtrade hyperopt-show --best -n -1 --print-json --no-header`